### PR TITLE
Integrate Democracy Pallet into Runtime and Configuration

### PR DESCRIPTION
**Pull request by asiniscalchi**
### **PR Type**
enhancement, configuration changes

___

### **Description**
- Integrated the `pallet_democracy` into the runtime, including its configuration and weights.
- Updated the collective council configuration with new voting proportion types.
- Added `pallet_democracy` to the list of benchmarks and weights.
- Updated `Cargo.toml` files to include `pallet-democracy` as a dependency.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><details><summary>5 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>benchmarks.rs</strong><dd><code>Include Democracy Pallet in Benchmarks</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

runtime/laos/src/benchmarks.rs

- Added `pallet_democracy` to the list of benchmarks.

</details>

  </td>
  <td><a href="https://github.com/freeverseio/laos/pull/759/files#diff-e1771479a1102a3e8954125c207e6f16ff6a00db135b75ea674751af1443fd1e">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>lib.rs</strong><dd><code>Integrate Democracy Pallet into Runtime</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

runtime/laos/src/lib.rs

- Added `Democracy` pallet to the runtime construction.

</details>

  </td>
  <td><a href="https://github.com/freeverseio/laos/pull/759/files#diff-003c87018a12c01c5266d5e1f3aba274f64cc6651fdb6b34abb88a6d1a10d073">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>mod.rs</strong><dd><code>Add Democracy Pallet to Weights Module</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

runtime/laos/src/weights/mod.rs

- Added `pallet_democracy` to the weights module.

</details>

  </td>
  <td><a href="https://github.com/freeverseio/laos/pull/759/files#diff-ba81c86b37ca206ece794d453027c58fd1ff8917fb1a04468ebe007768deb23b">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>pallet_collective.rs</strong><dd><code>Update Weights for Collective Pallet</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

runtime/laos/src/weights/pallet_collective.rs

- Updated autogenerated weights for `pallet_collective`.

</details>

  </td>
  <td><a href="https://github.com/freeverseio/laos/pull/759/files#diff-0377b3d6412af89360c4fbc275d1ff99f5ec0632851f478f4c1e2285bf9466b5">+56/-50</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>pallet_democracy.rs</strong><dd><code>Add Weights for Democracy Pallet</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

runtime/laos/src/weights/pallet_democracy.rs

- Added autogenerated weights for `pallet_democracy`.

</details>

  </td>
  <td><a href="https://github.com/freeverseio/laos/pull/759/files#diff-170b63d952de6f740f4b14b2e9d68f8ffbbac54abde8bbe64f97b5457011a400">+488/-0</a>&nbsp; </td>

</tr>                    
</table></details></td></tr><tr><td><strong>Configuration changes</strong></td><td><details><summary>4 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>collective_council.rs</strong><dd><code>Define Council Voting Proportion Types</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

runtime/laos/src/configs/collective_council.rs

- Defined new types for council voting proportions.

</details>

  </td>
  <td><a href="https://github.com/freeverseio/laos/pull/759/files#diff-31a89ba9921589c51af4b28017abbf3e0757f662a5c6d1f25c0227c6446cc1c0">+9/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>democracy.rs</strong><dd><code>Add Democracy Pallet Configuration</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

runtime/laos/src/configs/democracy.rs

<li>Added configuration for <code>pallet_democracy</code>.<br> <li> Defined parameter types and implemented the Config trait.<br>

</details>

  </td>
  <td><a href="https://github.com/freeverseio/laos/pull/759/files#diff-3be593d789e263668318328ecbb33bcf46ad31e297f9e43b4202e2201c97e674">+85/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>mod.rs</strong><dd><code>Add Democracy Module to Configurations</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

runtime/laos/src/configs/mod.rs

- Added `democracy` module to the configuration.

</details>

  </td>
  <td><a href="https://github.com/freeverseio/laos/pull/759/files#diff-450b756d1af9090dd62cc432c3c21a1a4f655d52ee8b477c3d81e1038434d68b">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>treasury.rs</strong><dd><code>Update Treasury Origins to Use Council Majority</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

runtime/laos/src/configs/treasury.rs

<li>Updated <code>ApproveOrigin</code> and <code>RejectOrigin</code> to use <code>CouncilMajority</code>.<br>

</details>

  </td>
  <td><a href="https://github.com/freeverseio/laos/pull/759/files#diff-094b4a74f7e7a446c70dd2ca710d6a7c87a1c46369ec9a8125621479e247219a">+1/-3</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></details></td></tr><tr><td><strong>Dependencies</strong></td><td><details><summary>2 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>Cargo.toml</strong><dd><code>Add Democracy Pallet Dependency</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

Cargo.toml

- Added `pallet-democracy` dependency.

</details>

  </td>
  <td><a href="https://github.com/freeverseio/laos/pull/759/files#diff-2e9d962a08321605940b5a657135052fbcef87b5e360662bb527c96d9a615542">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>Cargo.toml</strong><dd><code>Include Democracy Pallet in Workspace</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

runtime/laos/Cargo.toml

- Added `pallet-democracy` to workspace members and features.

</details>

  </td>
  <td><a href="https://github.com/freeverseio/laos/pull/759/files#diff-9b329717b96857f6884c9b9a32fc471267d6f9c0acc1a3b0e4758d73de6d34e5">+4/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></details></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: runtime/laos/src/configs/democracy.rs [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+	/// that voting stakers have an opportunity to remove themselves from the
+	/// system in the case where they are on the losing side of a vote.
+	type EnactmentPeriod = EnactmentPeriod;
+	/// A unanimous council can have the next scheduled referendum be a straight
+	/// default-carries (NTB) vote.
+	type ExternalDefaultOrigin = AllOfCouncil;
+	/// A simple-majority can have the next scheduled referendum be a straight
+	/// majority-carries vote.
+	type ExternalMajorityOrigin =
+		pallet_collective::EnsureProportionAtLeast<AccountId, CouncilCollective, 1, 2>;
```
Comment (magecnion) [inline runtime/laos/src/configs/democracy.rs:52]: comment above says _simple-majority_ whereas `pallet_collective::EnsureProportionAtLeast<AccountId, CouncilCollective, 1, 2>` is `HalfOfCouncil`

File: runtime/laos/src/configs/democracy.rs [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+	/// A straight majority of the council can decide what their next motion is.
+	type ExternalOrigin = HalfOfCouncil;
```
Comment (magecnion) [inline runtime/laos/src/configs/democracy.rs:54]: same as above

**Review by magecnion — APPROVED** [review]

File: runtime/laos/Cargo.toml [hunk 3]
```diff
@@ -149,6 +150,7 @@ std = [
 	"pallet-balances/std",
 	"pallet-elections-phragmen/std",
 	"pallet-session/std",
+	"pallet-democracy/std",
```
Comment (luispdm) [inline runtime/laos/Cargo.toml:153]: Not related to this line but the requested modification affects this file. We should replace
```toml
fast-mode=[]
```
for:
```toml
fast-runtime=[]
```
This is what `prod_or_fast` does: https://github.com/paritytech/polkadot-sdk/blob/a947cb831d8d56228e7aa84355410c50864a816e/polkadot/runtime/common/src/lib.rs#L272

**Review by luispdm — CHANGES_REQUESTED** [review]

**Review by magecnion — APPROVED** [review]

**Review by luispdm — APPROVED** [review]

**Pull request merged**